rollbar-android/src/main/java/com/rollbar/android/Rollbar.java

```diff
@@ -272,23 +293,28 @@ public Rollbar(Context context, String accessToken, String environment, boolean
       config = defaultConfig.build();
     }
 
-    if (config.sender() == null) {
-      DiskQueue queue = new DiskQueue.Builder()
-              .queueFolder(folder)
-              .build();
+    if (config.sender() != sender) {
```

Comment:
`sender` is the default sender created further up in this method. `config.sender` is the sender actually applied to the config. It may still be the default, or may have been updated by the `configProvider`. What this block does is, close the default sender we just created if it was removed from the config. 

The `!=` operator will compare the instance of the class, so this will detect if the new sender is  a new instance of the same class and still correctly close the default.